packages/ui/src/machines/actors/auth/signUp.ts

```diff
@@ -188,18 +188,20 @@ export const signUpActor = createMachine<SignUpContext, AuthEvent>(
           login_mechanisms,
         } = context;
 
-        const [primaryAlias] = login_mechanisms ?? ['username'];
+        const [primaryAlias = 'username'] = login_mechanisms;
 
         if (formValues.phone_number) {
-          formValues.phone_number = formValues.phone_number.replace(
-            /[^A-Z0-9+]/gi,
-            ''
-          );
+          formValues.phone_number =
+            `${formValues.country_code}${formValues.phone_number}`.replace(
+              /[^A-Z0-9+]/gi,
+              ''
+            );
         }
 
         const username = formValues[primaryAlias];
         delete formValues[primaryAlias];
         delete formValues.confirm_password; // confirm_password field should not be sent to Cognito
+        delete formValues.country_code;
```

Comment:
FYI, mutating this `formValues` object via `delete` is a sign we're assuming they're nearly 1-to-1 with `attribtues`.  That's not correct.

`attributes` (on line 209 below) should be created _from_ `formValues` (and any other data sources).

For example, what happens when signUp fails?  Will `formValues` still have the original `formValues.country_code`?

Reply:
It's a bit cheeky since `formValues` is created by spreading the remaining values from `context` after pulling out `password`. So it already is a (shallow) copy of the actual form values passed from the UI. We could be more explicit about what values we're pulling from `formValues`, though this is a bit less scalable. What do you think @ericclemmons?

Reply:
Ah, I see.  Yeah, I think I'm biased towards explicitness on knowing what the valid `attributes` even are.  IME, Forms aren't 1-to-1 with data models (Country code is a good example of this, as is confirm password).

I think I'm looking for an answer on _how do we know what valid attributes should even be sent_?  If we accept 16 different custom attributes, won't we end up doing something like `const attributes = pick(formValues, signUpAttributes)`?

Nothing to change here today, but just a callout that mutating an object can pose a risk and that we'll likely require a different solution when we support all custom attributes.